pkg/crawler/crawler.go

```diff
@@ -262,3 +320,239 @@ func (c *Crawler) fetchSHA1(url string) ([]byte, error) {
 	}
 	return sha1b, nil
 }
+
+func (c *Crawler) fetchAndSavePOMLicenseKeys(url string) ([]string, error) {
+	var keys []string
+	resp, err := c.http.Get(url)
+	if resp.StatusCode == http.StatusNotFound {
+		return keys, nil
+	}
+	if err != nil {
+		return keys, xerrors.Errorf("can't get pom xml from %s: %w", url, err)
+	}
+	defer resp.Body.Close()
+
+	var pomProject PomProject
+
+	decoder := xml.NewDecoder(resp.Body)
+	decoder.CharsetReader = charset.NewReaderLabel
+	err = decoder.Decode(&pomProject)
+
+	if err != nil {
+		return keys, xerrors.Errorf("can't parse pom xml from %s: %w", url, err)
+	}
+
+	if len(pomProject.Licenses) == 0 {
+		return keys, nil
+	}
+
+	for _, l := range pomProject.Licenses {
+		key := getLicenseKey(l)
+
+		// default value
+		l.NormalizedLicense = l.Name
+		l.LicenseKey = key
+
+		// update uniqueLicenseKeys map
+		c.uniqueLicenseKeys.Set(key, l)
```

Comment:
I think we don't need to calculate hash for license. We can use URL as unique identifier and skip `name` if URL exists.

Reply:
we can use hashes for license files created by URL.

Reply:
Intent behind saving not just unique license urls is : 

1. License can either be obtained from license url classification . If url is missing or classification failed, we fallback to the license name in POM.  Since the license classification is being done asynchronously, we need to track the license information tied to license keys so that same can be used during classification.
```
func getLicenseKey(l License) string {
	if len(l.URL) > 0 && strings.HasPrefix(l.URL, "http") {
		return hash(l.URL)
	}
	return hash(l.Name)
}
```

Reply:
okay, you are right. It make sense.
But i think we can remove `NormalizedLicense`. We can replace this field as logic for `name` + `url`, right?

Reply:
```
type License struct {
	Name              string `xml:"name"`
	URL               string `xml:"url"`
	NormalizedLicense string
	LicenseKey        string
}
```
Normalized license is to just differentiate the license information obtained from pom i.e Name and URl vs the license string that we finally intend to track. Since this struct is just used for processing and doesn't translate to any columns in the db, it shouldn't make a lot of difference in my opinion. Do you see any concerns with that.

Reply:
It looks as extra field. This can be confusing. Perhaps we need to move logic for this field to place where we get licenses from `uniqueLicenseKeys`? 

e.g. we got license from `uniqueLicenseKeys`:
if url -> use license classifier
if url doesn't exist or classifier didn't find licenses -> use name.

Reply:
Hmm. Got it. Let me give it a thought.

Reply:
Have got rid of the NormalizedLicense key. Thank you for pointing it out. Hope its cleaner now